client/X11/xf_cliprdr.c

```diff
@@ -1663,6 +1928,334 @@ static UINT xf_cliprdr_clipboard_file_range_failure(wClipboardDelegate* delegate
+	                                     nPositionHigh, size);
+}
+
+static void xf_cliprdr_fuse_lookup(fuse_req_t req, fuse_ino_t parent, const char* name)
+{
+	size_t index;
+	size_t count;
+	size_t child_ino;
+	BOOL found = FALSE;
+	struct fuse_entry_param e;
+	xfCliprdrFuseInode* parent_node;
+	xfCliprdrFuseInode* child_node;
+	xfClipboard* clipboard = (xfClipboard*)fuse_req_userdata(req);
+
+	ArrayList_Lock(clipboard->ino_list);
+	parent_node = (xfCliprdrFuseInode*)ArrayList_GetItem(clipboard->ino_list, parent - 1);
+	ArrayList_Unlock(clipboard->ino_list);
+	if (!parent_node || !parent_node->child_inos)
+	{
+		fuse_reply_err(req, ENOENT);
+		return;
+	}
+
+	ArrayList_Lock(parent_node->child_inos);
+	count = ArrayList_Count(parent_node->child_inos);
+	for (index = 0; index < count; index++)
+	{
+		child_ino = (size_t)ArrayList_GetItem(parent_node->child_inos, index);
+		ArrayList_Lock(clipboard->ino_list);
+		child_node = (xfCliprdrFuseInode*)ArrayList_GetItem(clipboard->ino_list, child_ino - 1);
+		ArrayList_Unlock(clipboard->ino_list);
+		if (child_node && strcmp(name, child_node->name) == 0)
+		{
+			found = TRUE;
+			break;
+		}
+	}
+	ArrayList_Unlock(parent_node->child_inos);
+	if (found == TRUE)
+	{
+		memset(&e, 0, sizeof(e));
+		e.ino = child_node->ino;
+		e.attr_timeout = 1.0;
+		e.entry_timeout = 1.0;
+		e.attr.st_ino = child_node->ino;
+		e.attr.st_mode = child_node->st_mode;
+		e.attr.st_nlink = 1;
+		e.attr.st_size = child_node->st_size;
+		e.attr.st_mtime = child_node->st_mtim.tv_sec;
+		fuse_reply_entry(req, &e);
+	}
+	else
+	{
+		fuse_reply_err(req, ENOENT);
+	}
+}
+
+static void xf_cliprdr_fuse_opendir(fuse_req_t req, fuse_ino_t ino, struct fuse_file_info* fi)
+{
+	xfCliprdrFuseInode* node;
+	xfClipboard* clipboard = (xfClipboard*)fuse_req_userdata(req);
+
+	ArrayList_Lock(clipboard->ino_list);
+	node = (xfCliprdrFuseInode*)ArrayList_GetItem(clipboard->ino_list, ino - 1);
+	ArrayList_Unlock(clipboard->ino_list);
+	if (!node)
+	{
+		fuse_reply_err(req, ENOENT);
+	}
+	else if ((node->st_mode & S_IFDIR) == 0)
+	{
+		fuse_reply_err(req, ENOTDIR);
+	}
+	else
+	{
+		fuse_reply_open(req, fi);
+	}
+}
+
+static struct fuse_lowlevel_ops xf_cliprdr_fuse_oper = {
+	.lookup = xf_cliprdr_fuse_lookup,
+	.getattr = xf_cliprdr_fuse_getattr,
+	.readdir = xf_cliprdr_fuse_readdir,
+	.open = xf_cliprdr_fuse_open,
+	.read = xf_cliprdr_fuse_read,
+	.opendir = xf_cliprdr_fuse_opendir,
+};
+
+static DWORD WINAPI xf_cliprdr_fuse_thread(LPVOID arg)
+{
+	xfClipboard* clipboard = (xfClipboard*)arg;
+
+	/* TODO: set up a filesystem base path for local URI */
+	/* TODO get basePath from config or use default*/
+	UINT32 basePathSize;
+	char* basePath;
+	char* tmpPath;
+	tmpPath = GetKnownPath(KNOWN_PATH_TEMP);
+	/* 10 is max length of DWORD string and 1 for \0 */
+	basePathSize = strlen(tmpPath) + strlen("/.xfreerdp.cliprdr.") + 11;
+	basePath = calloc(basePathSize, sizeof(char));
+	_snprintf(&basePath[0], basePathSize, "%s/.xfreerdp.cliprdr.%d", tmpPath,
```

Comment:
Using `mkdtemp` for the mount path should be safer here, since that will create a directory with a name that definitely won't collide with an existing one.